.golangci.yml

```diff
@@ -3,20 +3,115 @@ linters:
   enable:
     - deadcode
     - gocyclo
+    - revive
     - gosec
     - govet
     - ineffassign
     - misspell
     - staticcheck
     - errcheck
+    - unused
     - gosimple
     - varcheck
 
 linters-settings:
   gofmt:
     simplify: false
-  goimports:
-    local-prefixes: github.com/cosmos/cosmos-sdk
+  gosec:
+    excludes:
+    - G404 # G404: Use of weak random number generator (math/rand instead of crypto/rand)
+    - G306 # G306: Expect WriteFile permissions to be 0600 or less
+  revive:
```

Comment:
do we really need to enumerate these rules? from https://github.com/mgechev/revive#golangci-lint, seems the default behavior (drop-in replacement of golint) is good for us?

Reply:
some of the rules like `var-naming` and `line-length-limit` for example are not necessary for our project. However when I disable those rules, it disables all the other rules too, in other words when there is a `rules` field in the configuration it only follows whatever is set inside. And when i use `enable-all-rules: true`, it does the job but also removes all the default argument configurations for all enabled rules so I had to add them back for the applicable rules orz. so i've decided to put whatever is needed instead